plugins/dell-dock/fu-dell-dock-common.c

```diff
@@ -43,3 +43,19 @@ fu_dell_dock_set_power(FuDevice *device, guint8 target, gboolean enabled, GError
 
 	return fu_dell_dock_ec_modify_lock(parent, target, enabled, error);
 }
+
+const gchar *
+fu_dell_dock_get_instance_id(guint8 *type, DOCK_HWDEV_INFO *dev_list, guint16 *vid, guint16 *pid)
+{
+	for (guint i = 0; i < sizeof(dev_list); i++) {
```

Comment:
so with my suggestion for the instance_id sentinel you don't have to know the size of the list.  You can instead do something like this:

```
for (guint i = 0; dev_list[i] != NULL; i++)
```

Reply:
fixed in [271ccab](https://github.com/fwupd/fwupd/compare/4d3fcb65389ab9d212b973791292d7013e3dc56b..271ccabb94ec9bcc19e08b12d211150155e71d17#diff-2ba3278d75d6ee813fdeecbf19a971927c29dc6bf7b35cb470f845de3f049fe7R51)